openmdao/jacobians/assembled_jacobian.py

```diff
@@ -327,14 +333,90 @@ def _apply(self, d_inputs, d_outputs, d_residuals, mode):
         with system._unscaled_context(
                 outputs=[d_outputs], residuals=[d_residuals]):
             if mode == 'fwd':
-                d_residuals.iadd_data(int_mtx._prod(d_outputs.get_data(), mode, int_ranges))
-                if ext_mtx is not None:
-                    d_residuals.iadd_data(ext_mtx._prod(d_inputs.get_data(), mode, None))
+                if len(d_outputs._names) > 0 and len(d_residuals._names) > 0:
```

Comment:
I'd recommend just using 'if d_outputs._names and d_residuals._names:' instead of testing if len(x) > 0.  It's more concise and faster as well.

Reply:
Fixed